app/api.go

```diff
@@ -1,43 +1,45 @@
 package app
 
 import (
-	"context"
 	"fmt"
 	"net/http"
 	"strings"
 
-	"github.com/gorilla/mux"
 	"github.com/rs/zerolog/log"
 )
 
 const (
-	requestChallenges = "chals"
-	sessionCookie     = "haaukins_session"
+	requestedChallenges = "challenges"
+	sessionCookie       = "haaukins_session"
 )
 
 func (lm *LearningMaterialAPI) Handler() http.Handler {
-	m := mux.NewRouter()
-	m.HandleFunc("/api/{chals}", lm.request(lm.getOrCreateClient(lm.getOrCreateChallengeEnv())))
+	m := http.NewServeMux()
+	m.HandleFunc("/api/", lm.request(lm.getOrCreateClient(lm.getOrCreateChallengeEnv())))
 	m.HandleFunc("/admin/clients", lm.listClients())
 	m.HandleFunc("/admin/envs", lm.listEnvs())
+	m.HandleFunc("/guacamole/", lm.proxyHandler())
 	return m
 }
 
 func (lm *LearningMaterialAPI) request(next http.Handler) http.HandlerFunc {
 
 	return func(w http.ResponseWriter, r *http.Request) {
 
-		chals, err := lm.GetChallengesFromRequest(mux.Vars(r)["chals"])
+		// No need to sanitize the url requested
+		//https://stackoverflow.com/questions/23285364/does-go-sanitize-urls-for-web-requests
```

Comment:
is it enough ? @mrturkmencom

Reply:
Yeah, I guess. If anything goes wrong, we can check it anyway.